tests/node/test_node_picker.py

```
@@ -0,0 +1,151 @@
+# Helper function to handle common assertions
+def assert_node_attributes(node, expected_class, expected_attributes):
+    assert isinstance(node, expected_class)
+    for attr_name, expected_value in expected_attributes.items():
+        actual_value = getattr(node, attr_name)
+        assert actual_value == expected_value, f"Attribute {attr_name} expected {expected_value}, got {actual_value}"
+
+
+# Helper function to handle common exception checks
+def check_exception(node_type, kwargs, expected_exception, match):
+    with pytest.raises(expected_exception, match=match):
+        node = node_picker(node_type=node_type, **kwargs)
```

Ruff

Local variable `node` is assigned to but never used

Remove assignment to unused variable `node`

(F841)